class-components/src/app/layout.tsx

```diff
@@ -0,0 +1,25 @@
+'use client';
+
+import React from 'react';
+import { Provider } from 'react-redux';
+import '../styles/globals.css';
+import store from '../store/store';
+import { ThemeProvider } from '../contexts/ThemeContext';
+import TopSection from '../components/TopSection/TopSection';
+
+const RootLayout = ({ children }: { children: React.ReactNode }) => {
+  return (
+    <Provider store={store}>
+      <ThemeProvider>
+        <html lang='en'>
+          <body>
+            <TopSection />
+            <main>{children}</main>
+          </body>
+        </html>
+      </ThemeProvider>
+    </Provider>
+  );
+};
```

Comment:
You should probably import these from Next and use them instead of <html>, etc
`import { Html, Main} from 'next/document';`